app/src/main/java/com/bnyro/translate/api/deepl/DeeplEngine.kt

```diff
@@ -33,43 +48,124 @@ class DeeplEngine : TranslationEngine(
 ) {
     val useFreeApiKey = this.name + "selectedApiVersion"
     private val nonFreeUrl = "https://api.deepl.com"
+    private val webUrl = "https://www2.deepl.com"
     private lateinit var api: DeepL
     override fun createOrRecreate(): TranslationEngine = apply {
         api = RetrofitHelper.createApi(this)
     }
 
     private val apiKeyString = "DeepL-Auth-Key " + getApiKey()
 
-    override suspend fun getLanguages(): List<Language> = api.getLanguages(
-        apiKeyString
-    ).map {
-        Language(
-            code = it.language.lowercase(),
-            name = it.name
-        )
-    }.toMutableList().apply {
-        add(
-            0,
+    override suspend fun getLanguages(): List<Language> = if (getApiKey().isNotEmpty()) {
```

Comment:
```suggestion
    override suspend fun getLanguages(): List<Language> = listOf(Language("", "Auto")) + if (getApiKey().isNotEmpty()) {
```
That's not a perfect way to do this either, but it makes the code more readable looking at line 67-75 (and 78)